lib/run-compilation.js

```diff
@@ -49,5 +55,8 @@ module.exports = function runCompilation(options, compilation, done) {
     errors.forEach(function (err) {
       compilation.errors.push(err);
     });
+    warnings.forEach(function (warning) {
+      compilation.warnings.push(warning);
```

Comment:
Please use the `.concat` method instead of going through each warning. I've been meaning to fix the original code as well so if you could change it that'd be great.